include/swift/AST/Identifier.h

```diff
@@ -395,6 +420,16 @@ class DeclName {
   StringRef getString(llvm::SmallVectorImpl<char> &scratch,
                       bool skipEmptyArgumentNames = false) const;
 
+  /// Return the string representation of the name
+  StringRef str() const {
```

Comment:
This is just not correct; it drops the compound part on the floor. Please remove these.